.github/workflows/rust.yml

```diff
@@ -62,6 +62,16 @@ jobs:
       - run: cargo fmt --all -- --check
       - run: cargo clippy --all-targets --all-features -- -D warnings
 
+  docs:
+    name: Check for documentation errors
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - run: cargo doc --locked --all-features --no-deps --document-private-items
```

Comment:
This is [failing in CI](https://github.com/djc/instant-acme/actions/runs/15828801147/job/44615908871?pr=112) because of `--locked`.

I had a couple small CI changes I wanted to suggest (matching the push branch pattern w/ rustls org, cancelling in-progress runs) so I made an alternative PR that checks in the `Cargo.lock` & switches to `--locked` elsewhere ahead of this commit adding docs coverage.